File: src/main/java/com/back/domain/auction/auction/dto/response/AuctionDetailResponse.java
package com.back.domain.auction.auction.dto.response;

import com.back.domain.auction.auction.entity.Auction;
import com.back.domain.auction.auction.entity.AuctionStatus;
import com.back.domain.auction.auction.entity.CancellerRole;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;

@Getter
@NoArgsConstructor
public class AuctionDetailResponse {
    private Integer auctionId;
    private String name;
    private String description;
    private Integer startPrice;
    private Integer currentHighestBid;
    private Integer buyNowPrice;
    private Integer bidCount;
    private AuctionStatus status;
    private LocalDateTime startAt;
    private LocalDateTime endAt;
    private List<String> imageUrls;
    private SellerDto seller;
    private String categoryName;

    private Integer winnerId;
    private LocalDateTime closedAt;

    private Integer cancelledBy;
    private CancellerRole cancellerRole;
    private String cancellerRoleDescription;

    public AuctionDetailResponse(Auction auction) {
        this.auctionId = auction.getId();
        this.name = auction.getName();
        this.description = auction.getDescription();
        this.startPrice = auction.getStartPrice();
        this.currentHighestBid = auction.getCurrentHighestBid();
        this.buyNowPrice = auction.getBuyNowPrice();
        this.bidCount = auction.getBidCount();
        this.status = auction.getStatus();
        this.startAt = auction.getStartAt();
        this.endAt = auction.getEndAt();
        this.categoryName = auction.getCategory().getName();

        this.winnerId = auction.getWinnerId();
        this.closedAt = auction.getClosedAt();

        this.cancelledBy = auction.getCancelledBy();
        this.cancellerRole = auction.getCancellerRole();
        this.cancellerRoleDescription = auction.getCancellerRole() != null
                ? auction.getCancellerRole().getDescription()
                : null;

        this.imageUrls = auction.getAuctionImages().stream()
                .map(auctionImage -> auctionImage.getImage().getUrl())
                .collect(Collectors.toList());

        Double reputationScore = null;
        if (auction.getSeller().getReputation() != null) {
            reputationScore = auction.getSeller().getReputation().getScore();
        }

        this.seller = new SellerDto(
                auction.getSeller().getId(),
                auction.getSeller().getNickname(),
                reputationScore
        );
    }
}

File: src/main/java/com/back/domain/auction/auction/dto/response/AuctionDetailResponse.kt
package com.back.domain.auction.auction.dto.response

import com.back.domain.auction.auction.entity.Auction
import com.back.domain.auction.auction.entity.AuctionStatus
import com.back.domain.auction.auction.entity.CancellerRole
import java.time.LocalDateTime

class AuctionDetailResponse() {
    var auctionId: Int? = null
    var name: String? = null
    var description: String? = null
    var startPrice: Int? = null
    var currentHighestBid: Int? = null
    var buyNowPrice: Int? = null
    var bidCount: Int? = null
    var status: AuctionStatus? = null
    var startAt: LocalDateTime? = null
    var endAt: LocalDateTime? = null
    var imageUrls: List<String?> = emptyList()
    var seller: SellerDto? = null
    var categoryName: String? = null

    var winnerId: Int? = null
    var closedAt: LocalDateTime? = null

    var cancelledBy: Int? = null
    var cancellerRole: CancellerRole? = null
    var cancellerRoleDescription: String? = null

    constructor(auction: Auction) : this() {
        auctionId = auction.id
        name = auction.name
        description = auction.description
        startPrice = auction.startPrice
        currentHighestBid = auction.currentHighestBid
        buyNowPrice = auction.buyNowPrice
        bidCount = auction.bidCount
        status = auction.status
        startAt = auction.startAt
        endAt = auction.endAt
        categoryName = auction.category.name

        winnerId = auction.winnerId
        closedAt = auction.closedAt

        cancelledBy = auction.cancelledBy
        cancellerRole = auction.cancellerRole
        cancellerRoleDescription = auction.cancellerRole?.description

        imageUrls = auction.auctionImages.map { it.image.url }

        val reputationScore = auction.seller.reputation?.score
        seller = SellerDto(
            auction.seller.id,
            auction.seller.nickname,
            reputationScore
        )
    }
}
